web/crux/src/app/registry/registry.http.controller.ts

```diff
@@ -1,28 +1,85 @@
 import { Identity } from '@ory/kratos-client'
+import { Delete, Post, Put } from '@nestjs/common/decorators/http/request-mapping.decorator'
+import { Body, Param } from '@nestjs/common/decorators/http/route-params.decorator'
 import JwtAuthGuard from '../token/jwt-auth.guard'
 import RegistryService from './registry.service'
+import {
+  CreateRegistryRequest,
+  RegistryDetailsResponse,
+  RegistryListResponse,
+  UpdateRegistryRequest,
+} from './registry.dto'
+import DeleteRegistryValidationPipe from './pipes/registry.delete.pipe'
+import UpdateRegistryInterceptor from './interceptors/registry.update.interceptor'
 
-@Controller('registry')
-@UseGuards(JwtAuthGuard)
-@UseInterceptors(HttpLoggerInterceptor, PrismaErrorInterceptor, HttpIdentityInterceptor)
-@UseFilters(HttpExceptionFilter)
+@Controller('registries')
+@ApiTags('registries')
+@UsePipes(
+  new ValidationPipe({
+    // TODO(@robot9706): Move to global pipes after removing gRPC
+    transform: true,
+  }),
+)
+@UseInterceptors(HttpLoggerInterceptor, HttpIdentityInterceptor, PrismaErrorInterceptor)
 export default class RegistryHttpController {
   constructor(private service: RegistryService) {}
 
   @Get()
-  @ApiBody({ type: AccessRequestDto })
-  @ApiOkResponse({ type: RegistryListResponseDto })
-  @AuditLogLevel('disabled')
+  @UseGuards(JwtAuthGuard) // TODO(@robot9706): Move to controller when crux.ts getRegistryDetails is fixed
+  @AuditLogLevel('disabled') // TODO(@robot9706): Refactor the auditlog after removing gRPC
   async getRegistries(@IdentityFromRequest() identity: Identity): Promise<RegistryListResponse> {
     return await this.service.getRegistries(identity)
   }
+
+  @Get('/:id')
+  // TODO(@robot9706): Refactor the auditlog after removing gRPC
+  // TODO (robot9706): Access check?
+  @AuditLogLevel('disabled')
+  async getRegistry(@Param('id') id: string): Promise<RegistryDetailsResponse> {
+    return await this.service.getRegistryDetails(id)
+  }
+
+  @Post()
+  @UseGuards(JwtAuthGuard) // TODO(@robot9706): Move to controller when crux.ts getRegistryDetails is fixed
+  @AuditLogLevel('disabled')
+  async createRegistry(
+    @Body() request: CreateRegistryRequest,
+    @IdentityFromRequest() identity: Identity,
+    @Response() res: Res,
+  ): Promise<void> {
+    const registry = await this.service.createRegistry(request, identity)
+
+    res.set('Location', `/registries/${registry.id}`).status(201).json(registry)
+  }
+
+  @Put('/:id')
+  @UseGuards(JwtAuthGuard) // TODO(@robot9706): Move to controller when crux.ts getRegistryDetails is fixed
+  @AuditLogLevel('disabled')
+  @UseInterceptors(UpdateRegistryInterceptor)
+  async updateRegistry(
+    @Param('id') id: string,
+    @Body() request: UpdateRegistryRequest,
+    @IdentityFromRequest() identity: Identity,
+    @Response() res: Res,
+  ): Promise<void> {
+    await this.service.updateRegistry(id, request, identity)
```

Comment:
When updating a resource we wan't do add the location too, no? Any opinion @m8vago ?

Reply:
I tought about it too. The article m8 linked in the ticket did not say anything about this though.

Reply:
Talk about it in person with the team.

Reply:
Leaving it like this, as we've discussed in person.